plugins/torrent/deluge_

```diff
@@ -0,0 +1,399 @@
+            interesting_status = [
+                'upload_rate', 'payload_upload_rate',
+                'download_rate', 'payload_download_rate']
+            client.core.get_session_status(interesting_status).addCallbacks(
+                self.on_bandwidth,
+                self.end_session,
+                errbackArgs=("get_session_status failed"))
+        elif self.mode == "states":
+            log.debug("Calling get_session_state")
+            client.core.get_session_state().addCallbacks(
+                self.on_session_state,
+                self.end_session,
+                errbackArgs=("get_session_state failed"))
+
+    def on_num_connections(self, num_connections):
+        log.debug("Got num_connections from the daemon : {0}".format(
+            num_connections))
+        print("{0}.value {1}".format(
+            names_for_munin["numConnections"], num_connections))
+        self.end_session("Done")
+
+    def on_bandwidth(self, values):
+        log.debug(
+            "Got bandwidth info from the daemon : {0}".format(values))
+
+        download_rate = values['download_rate']
+        payload_download_rate = values['payload_download_rate']
+        overhead_download_rate = download_rate - payload_download_rate
+        upload_rate = values['upload_rate']
+        payload_upload_rate = values['payload_upload_rate']
+        overhead_upload_rate = upload_rate - payload_upload_rate
+
+        print("{0}.value {1}".format(
+            names_for_munin["payloadDownloadRate"], payload_download_rate))
+        print("{0}.value {1}".format(
+            names_for_munin["overheadDownloadRate"], overhead_download_rate))
+        print("{0}.value {1}".format(
+            names_for_munin["payloadUploadRate"], payload_upload_rate))
+        print("{0}.value {1}".format(
+            names_for_munin["overheadUploadRate"], overhead_upload_rate))
+        self.end_session("Done")
+
+    def on_session_state(self, torrent_ids):
+        log.debug("Got session state from the daemon")
+
+        self.states = {}
+        for state_name in torrent_states:
+            self.states[state_name] = 0
+
+        deferred_list = []
+        for torrent_id in torrent_ids:
+            log.debug(" - TorrentId : {0}".format(torrent_id))
+            d = client.core.get_torrent_status(torrent_id, ['state'])
+            d.addCallback(self.on_one_torrent_info, torrent_id)
+            d.addErrback(self.on_one_torrent_info_failed, torrent_id)
+            deferred_list.append(d)
+
+        defer.DeferredList(deferred_list).addCallback(
+            self.on_all_torrent_info_fetched)
+
+    def on_one_torrent_info_failed(self, torrent_id):
+        log.debug("Failed fetching torrent info {0}".format(torrent_id))
+        self.state["Error"] = self.state["Error"] + 1
+
+    def on_one_torrent_info(self, value, torrent_id):
+        log.debug("Got torrent info : {0} -> {1}".format(torrent_id, value))
+        if 'state' in value:
```

Comment:
A maybe more concise and readable alternative could be: `state = value.get("state", "Error")`.